text/0000-tracked-properties.md

```diff
@@ -0,0 +1,1029 @@
+  }
+}
+```
+
+The reverse, however, is not true - computed properties will be able to add
+tracked properties, and listen to dependencies explicitly. In some cases, this
+may be preferable, though tracked getter should be the conventional standard
+with the long term goal of removing all explicit dependencies.
+
+## How we teach this
+
+There are three different aspects of tracked properties which need to be
+considered for the learning story:
+
+1. **General usage.** Which properties should I mark as tracked? How do I
+   consume them? How do I trigger changes?
+2. **Interop with classic systems.** How do I safely consume tracked properties
+   from classic classes and computeds? How do I safely consume classic APIs from
+   tracked properties?
+3. **Interop with non-Ember systems.** How do I tell my app that something has
+   changed in MobX objects, RxJS objects, Redux, etc.
+
+### General Usage
+
+The mental model with tracked properties is that anything _mutable_ should be
+tracked. If a value will ever change, it should have the `@tracked` decorator
+attached to it.
+
+After that, usage should be "Just Javascript". You can safely access values
+using any syntax you like, including desctructuring, and you can update values
+using standard assignments.
+
+```js
+// Dot notation
+const fullName = person.fullName;
+// Destructuring
+const { fullName } = person;
+// Bracket notation for computed property names
+const fullName = person['fullName'];
+
+// Simple assignment
+this.firstName = 'Yehuda';
+// Addition assignment (+=)
+this.lastName += 'Katz';
+// Increment operator
+this.age++;
+```
+
+#### Triggering Updates on Complex Objects
+
+There may be cases where users want to update values in complex, untracked
+objects such as arrays or POJOs. `@tracked` will only be usable with class
+syntax at first, and while it may make sense to formalize these objects into
+tracked classes in some cases, this will not always be the case.
+
+To do this, users can re-set a tracked value directly after its inner values
+have been updated.
+
+```js
+class SomeComponent extends Component {
+  @tracked items = [];
+
+  @action
+  pushItem(item) {
+    let { items } = this;
+
+    items.push(item);
+
+    this.items = items;
+  }
+}
+```
+
+This may seem a bit strange at first, but it allows users to mentally scope
+off a tree of objects. They manipulate internals as they see fit, and the only
+operation they need to do to update state is set the nearest tracked property.
+
+### Interop with Classic Systems
+
+There are two cases that we need to consider when teaching interoperability:
+
+1. Tracked getters accessing non-tracked properties and computeds
+2. Computed getters accessing tracked properties
+
+In the first case, the general rule of thumb is to use `get` if you want to be
+100% safe. In cases where you are certain that the values you are accessing are
+tracked, computeds, or immutable, you can safely standard access syntax.
```

Comment:
```suggestion
tracked, computeds, or immutable, you can safely use standard access syntax.
```

Reply:
Also, maybe add a link referring back to "**`get` and `set`**" to explain _why_.